### DISCUSSION TITLE
bug: custom rpc requests eip-1193 gets sent to the RPC provider instead of to the wallet with web3modal

### DISCUSSION THREAD

**Discussion by Sednaoui** [wevm/wagmi#2755] [Ideas]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Package Version

1.3.3

### Current Behavior

Hello,

I am looking to send a custom rpc request to a smart contract wallet wallet connected through web3modal / walletconnect. I am using the hook `useWalletClient` . 

The error happens then `useWalletClient` sends the custom requests the default rpc provider instead of to the wallet. So the request is being rejected with the following error, when connected with a wallet through walletconnect. 

![image](https://github.com/wagmi-dev/wagmi/assets/7014833/f0575534-0246-4aa8-bb66-a40d8be33c43)

I have tried metamask mobile, rainbow mobile, and candide wallet.  

### Expected Behavior

The expected behavior is to send the custom request to the wallet directly. Using metamask browser injected provider, the request is sent to the wallet directly, and the error is correct as expected:

![image](https://github.com/wagmi-dev/wagmi/assets/7014833/07feebb3-1723-426a-9f37-909ec02c77d9)

### Steps To Reproduce

```js
const customRequest = {
    method: "wallet_customRequest",
    params: [..]
}
```
`useWalletClient()` hook 

```js
import { useWalletClient } from 'wagmi"

const { data: walletClient } = useWalletClient();
..
await walletClient.request(customRequest);
```

### Link to Minimal Reproducible Example (StackBlitz, CodeSandbox, GitHub repo etc.)

https://github.com/Sednaoui/wagmi-web3modal-custom-rpc-error

### Anything else?
I am testing on optimism network.

_No response_

**Comment by tmm** [comment]
At the moment, the WalletConnect Connector does not support adding additional methods so only the [ones defined in the `init` function](https://github.com/wagmi-dev/references/blob/main/packages/connectors/src/walletConnect.ts#L302) are valid. Moving to feature request.

**Reply by joanseg** [reply]
Hi @tmm thanks for your feedback. Please, could you let us know what is the best way to reach Wagmi team in order to explain what we are trying to achieve, so you have all the information required to prioritize it?
Thanks for your help.

**Reply by tmm** [reply]
Feel free to drop some info on what you are looking to accomplish here. (FYI we only prioritize feature requests when they seem of general interest to the community.)

**Reply by Sednaoui** [reply]
Thanks @tmm for the clarification. We are personally trying to integrate [EIP-5792](https://eips.ethereum.org/EIPS/eip-5792) for batch transactions. TWith smart contract wallets now getting popular, dapp frontend can now communicate JSON-RPC method for sending multiple function calls from the user’s wallet. Dapps are able to offer one-click experience for multi-step interactions of bundled calls. The following methods are the ones we are using: 

- `wallet_sendFunctionCallBundle`
- `wallet_getBundleStatus`
- `wallet_showBundleStatus`

I can't speak for the general community. But with this bug open, I believe no one can experiment with new ERCs for the ethereum provider API with wagmi library and web3modal.

**Reply by joanseg** [reply]
Adding to @Sednaoui . 
A concrete example we want to achieve is to Approve, and confirm payment in the same call.
This way we achieve same experience as tradicional payment methods.
We will also add the call to the paymaster to sponsor transaction.
This will enable any Dapp that uses Wagmi and Web3modal (or Rainbowkit,...) To create experiences with less friction for the user, and enable mass adoption.
Please let us know if your team can review and prioritise and let us know, without any commitment, if will be done at all, short or long term.
We use extensively Wagmi and Viem now💪. And we want to avoid using ethers.js work arounds in our code😅.
Many thanks for your help and attention beforehand🙏.
Cheers.

**Reply by joanseg** [reply]
Dear @tmm, any update on this? We are willing to collaborate and support this request. If needed, we can help to develop or pay for it.
Thanks for your support in advance.
